pkg/recipes/terraform/execute_test.go

```diff
@@ -157,19 +158,29 @@ func TestSetEnvironmentVariables(t *testing.T) {
 			},
 		},
 	}
-
 	for _, tc := range testCase {
 		t.Run(tc.name, func(t *testing.T) {
 			ctx := testcontext.New(t)
-			workingDir := t.TempDir()
 
-			tf, err := tfexec.NewTerraform(workingDir, filepath.Join(workingDir, "terraform"))
-			require.NoError(t, err)
+			// Get the environment variables to set
+			envVars := tc.opts.EnvConfig.RecipeConfig.Env.AdditionalProperties
 
 			e := executor{}
 
-			err = e.setEnvironmentVariables(ctx, tf, tc.opts.EnvConfig)
-			require.NoError(t, err)
+			for key, expectedValue := range envVars {
+				err := os.Unsetenv(key)
+				require.NoError(t, err)
+
+				err = e.setEnvironmentVariables(ctx, tc.opts.EnvConfig)
```

Comment:
This function call should be outside the loop. With current implementation we are calling the same function with same inputs for every env var in recipe config. The loop is only needed for validation.

Reply:
yikes! code reverted to original code where this call is still valid.